src/datasets/containers.jl

```diff
@@ -1,4 +1,216 @@
+
+                        if is_first_case
+                            num_dimensions = length(dimensions)
+
+                            if class_labels
+                                num_dimensions -= 1
+                            end
+                            
+                            # for _dim in 1:num_dimensions
+                            #     push!(instance_list, [])
+                            # end
+
+                            is_first_case = false
+                        end
+                        
+                        # See how many dimensions that the case whose data
+                        # in represented in this line has
+
+                        this_line_num_dim = length(dimensions)
+
+                        if class_labels
+                            this_line_num_dim -= 1
+                        end
+
+                        # All dimensions should be included for all series,
+                        # even if they are empty
+
+                        if this_line_num_dim != num_dimensions
+                        end
+
+                        # Process the data for each dimension
+
+                        arr = Array{Float32, 2}(undef, num_dimensions, series_length)
+
+                        for dim in 1:num_dimensions
+                            dimension = strip(dimensions[dim], ',')
+
+                            if !isempty(dimension)
+                                data_series = split(dimension, ',')
+                                data_series = [parse(Float32, i) for i in data_series]
+                                arr[dim, 1:end] = data_series
+                                # println(data_series)
+                                # data_series = [parse(Float32, i) for i in data_series]
+                                # push!(instance_list[dim], data_series)
+                            else
+                                tmp = Array{Float32, 1}(undef, 100)
+                                arr[dim, 1:end] = tmp
+                                # push!(instance_list[dim], [])
+                            end
+                        end
+
+                        push!(instance_list, arr)
+
+                        push!(class_val_list, strip(dimensions[num_dimensions+1], ' '))
+
+                    end
+                end
+            end
+            line_num += 1
+        end
+    end
+
+    if line_num > 0
+
+        num_samples = length(instance_list)
+        series_length = size(instance_list[1])[2]
+
+        data = Array{Float32, 3}(undef, num_samples, num_dimensions, series_length)
+
+        for sample in 1:num_samples
+            data[sample, 1:end, 1:end] = instance_list[sample]
+        end
+
+        # Check if we should return any associated class labels separately
+        if class_labels
+            return data, class_val_list
+        else
+            return data
+        end
```

Comment:
Returning one or two things based on a conditional is a little surprising. Consider either always returning class_val_list and/or making the second return value some meaningful null value (nothing, empty array, whatever makes the most sense).